### ISSUE TITLE
l10n_es

### ISSUE THREAD

**Issue by lap-odoo** [odoo/odoo#18565]
Dear Odoo Partners,

At Odoo, a lot of effort is put into improving the accounting localizations for each country and also to maintain these accounting localizations up-to-date. To achieve that, we will need your help.

One github issue will be created per accounting localization (e.g. l10n_be for Belgium). We expect from you to make suggestions about what should be improved in the specific accounting localization. Do not hesitate to share files with detailed explanations.

As soon as changes will be implemented, you will be informed and will have the opportunity to test it and share your feedback.

This github issue will also be useful to keep you informed about all the ongoing improvements concerning a specific accounting localization.

**Comment by lap-odoo** [comment]
Dear Partners, 

For now, we are working on: 
- Improving the CoA
- Working on the AET reports : models 111, 115, 303, 347, 349 and AEAT SII

I'll inform you as soon as this can be tested. 

Do you think we should also work on the following AEAT models: 180, 190, 200, 202, 390 ? If yes, do you have where we could find clear specifications about these reports? 

Thank you for your help

**Comment by pedrobaeza** [comment]
Hi, Laura,

I'm Pedro, current OCA Spanish Localization PSC leader. I'm glad to see that you're pushing the improvement of the localizations.

I have been talking previously with @qdp-odoo about this improvement, so my first question is: are you aware of such conversations?

Next question is, and I need this to be clearly defined, who are the previous actors that have told you about requirements. I see about SII and other models that I haven't talked with Quentin, so someone more have been involved meanwhile. We need to be all in the same wave and use public channels for joining the requirements.

And the last question is: how are you going to implement these things? Are there any development branch that I can check?

**Comment by liebana** [comment]
Why rebuild again everything when we have those developments in our community and public branch? What's wrong on joining forces and improve what it's already done? Do you really think you will be able to maintain an "up to date" localization for each country?

Let me tell you what it's going to happen: you will deliver some enterprise shitty modules that will bring new incompatibilities with the community ones, no one will use them except the same shitty partners that are pushing this because they are not capable of taking any lead role on the spanish community.

**Comment by lap-odoo** [comment]
@pedrobaeza Hi Pedro, yes I'm aware of your discussions with @qdp-odoo , most of the specs are based on this. We received feedback from other partners but also from our internal departments. The main goal of this Github issue is to make sure everyone is on the same page regarding the improvements that will be done on Spanish Accounting Localization. 

@oco-odoo will inform you as soon as the branch will be ready to test. 

@liebana We don't plan to rebuild anything but to improve what we have. And you won't be forced to use our shitty modules.

**Comment by js-landoo** [comment]
Hi Laura, @lap-odoo.
Great idea to publish an issue to share your roadmap and to ask for colaborations.
With CoA, I assume you are planning to update taxes. Actual ones in your l10n_es module are deprecated.
Answering to your question:
- AEAT models: 180, 190, 200, 202. Are not interesting.
- AEAT models: 390 is annual version of 303. It would be great if you develop it.
Thanks for your efforts.

**Comment by oco-odoo** [comment]
@js-landoo, when you say the taxes in our module are deprecated, do you mean the ones defined in here (https://github.com/OCA/l10n-spain/blob/8.0/l10n_es/data/taxes_common.xml) are too ? From the quick look I had at it, it is the same file as the one we already had in 8.0, isn't it ? The most recent version of our code already had modified the taxes defined in l10n_es (see master branch here https://github.com/odoo/odoo/blob/master/addons/l10n_es/data/account_tax_data.xml), so are these taxes correct, or do they still correspond to some old version of the regulation ? If you have any clear list of the taxes currently in use, feel free to share it, of course !

**Comment by antonylesuisse** [comment]
test

**Comment by pedrobaeza** [comment]
@oco-odoo there are still missing taxes in your updated branch, like https://github.com/OCA/l10n-spain/blob/8.0/l10n_es/data/taxes_common.xml#L1034-L1051

**Comment by js-landoo** [comment]
@oco-odoo I was speaking about 10.0 branch. This one looks updated.
But about taxes in OCA/l10n-spain, @pedrobaeza has better knowledge.

**Comment by oco-odoo** [comment]
@pedrobaeza @js-landoo Alright, I'll update the taxes to reflect the ones on the branch Pedro gave. Thanks for your help !

**Comment by pedrobaeza** [comment]
@oco-odoo Thanks! Let me know when you have it if you want me to check it.

**Comment by oco-odoo** [comment]
@pedrobaeza, some additional question : in mod 303 report, your module splits casilla 14 (Modificación bases y cuotas) in two parts, one for purchases and the other for sales. Is it some design choice (if so, what's the reason ?), or I am missing some part of the regulation instructing to do so ?

**Comment by pedrobaeza** [comment]
@oco-odoo that's because the formula for getting the amounts varies depending if they are sales or purchases. As our tax calculation engine only allows one type of computation by line, we thought it was better to allow multiple lines with the same number and sum them than to create a submodel for specifying this.

Any way, this is very important: __you can't get the 303 report using only tags and obtaining the balance of the accounts tagged with them__, as you can see with these "casillas" (numbered fields). Sometimes you only get debit amounts, other credit amounts, you inverse the sign in the result... so you will need a similar engine.

**Comment by oco-odoo** [comment]
@pedrobaeza Ok, now I see. Thanks. As for the fact using only tags in the usual way will not work as expected, I was aware of that and indeed already developed a "home-made" solution therefore. So, no worry ;)

**Comment by pedrobaeza** [comment]
OK, let me know any other doubt.

**Comment by oco-odoo** [comment]
@pedrobaeza , the L10n_es_aeat_mod115 module of your 10.0 branch still uses v8 API, and the tax codes it uses to find the move lines for casillas 02 and 03 do not seem to be linked to any actual tax in taxes_common.xml (from l10n_es), so I can't tell which taxes should be taken into account for these two. Can you help me ?

**Comment by pedrobaeza** [comment]
That's because is not yet ported. See in the manifest that the key `installable` is set to `False`. FYI, we have opened issues for tracking the progress on the migrations. This one is for v9: https://github.com/OCA/l10n-spain/issues/225, and this one is for v10: https://github.com/OCA/l10n-spain/issues/401. On v9, the module is already migrated to new API and new tax system, and merged: https://github.com/OCA/l10n-spain/tree/9.0/l10n_es_aeat_mod115, and for v10 there's a PR, but still WIP as the CIs are red (https://github.com/OCA/l10n-spain/pull/490). You can take 9.0 version as reference any way.

**Comment by oco-odoo** [comment]
Ok, thanks !

**Comment by oco-odoo** [comment]
@pedrobaeza : Small question about the balance sheets in your module. Some of their lines are created without any way of computing a value for them, and will hence always be 0. Is it some legal requirement or so ? I suppose I must do the same ? As an example, see line with code 11700 ("VII. Deudores comerciales no corrientes") in l10n_es_account_balance_report/data/balance_abreviado.xml

**Comment by pedrobaeza** [comment]
Yeah, that one is a special case. If you see the comment, it refers to a section of the documentation, as there was not an specific accounts for computing it. They are barely used (I don't have any customer that needs that section). Check this link https://www.iberley.es/temas/norma-6-balance-neca-6-59373 the 8th point to decide if you want to fix an account for checking this or leave it un-computed (usually, Spanish accountants create an specific account for each bank credit and without common prefix - 520, 521..., so it makes it more difficult to set a rule).
*

**Comment by oco-odoo** [comment]
Ok, thank for your help again. I think we will manage the variation of code definitions for those accounts with tags, so that they still do appear clearly into the balance sheet if defined.

Some more question: here https://github.com/OCA/l10n-spain/blob/10.0/l10n_es/data/account_type.xml , you define several account types, but you do not seem to explicitly use them anywhere else in your code. So, I assume they are just there for sorting purposes ? We would like to avoid creating new account types if not absolutely necessary, so would you have a suggestion of existing account types that would correspond to those ? From what I could tell by comparing your files with the version on our community module, types account_type_capital and account_type_stock both seem to correspond to our data_account_type_current_assets type, correct ?

**Comment by jbeficent** [comment]
@lap-odoo is this going to land on Odoo CE or Enterprise? I think that it is fair to be clear here. @pedrobaeza and the spanish localization team has been promoting an open version for years, and if you accept their feedback it is fair that the resulting work should be in CE

**Comment by pedrobaeza** [comment]
@jbeficent Indeed I'm a bit disappointed with the move of certain account features to enterprise (https://github.com/odoo/odoo/pull/18589), - which seems to be only the first step (assets and budgets will be next) -, and I have mixed feelings about continuing contributing to this issue.

But I think that `l10n_es` will continue being in community. If not, of course Odoo should warn us. This module includes correct charts of accounts, updated taxes and meta-data like account/tax tags. I think it's good for both Community and Odoo Enterprise to unify this module and don't need to maintain an specific version of `l10n_es` in OCA/l10n-spain. That's my main motivation for contributing here.

The trick/trap here is on the included "meta-data", that will be only directly exploitable on account reporting engine for enterprise, which will include taxes report and its specific computations needed for AEAT (probably embedded on the existing `l10n_es_reports` module). Our OCA/l10n-spain `l10n_es_aeat*` modules will continue working with the same base data of course, but no other benefit as we won't be able to use Odoo's enterprise engine for computing AEAT fields.

On the contrary, Odoo is looking at our AGPL code for reimplementing the requirements in their private code. That reduces a lot the amount of work to do as the knowledge is already put there (and that knowledge is for what a lot of accounting service companies charge fees). Is that legal? Probably yes, because there's no direct derivative nor dependent work, although I remember a clause in enterprise agreement saying that we can't look at enterprise code for reimplementing features contained in it. Is that ethic? It's indeed more debatable as they are privatizing an open knowledge.

Since this, I need the clarification about `l10n_es` for continuing answering questions, and my code review will be limited to the community PRs.

**Comment by anajuaristi** [comment]
+1 @pedrobaeza. I agree everything you say. I feel same way.
@jbeficent It should be on CE but I'm afraid it will be not. Our several years work will be privaticed. I hope I'm mistaken.  It would be great that I had to recognice my mistake.

**Comment by lasley** [comment]
I don't see any reason for the community to put forth effort into a solution that is not available to them. They can legally use our code for inspiration, but I don't see a reason to feed it to them personally.

**Comment by lap-odoo** [comment]
Hi all, 

You can find below an explanation of the reasons for these changes and their implications : 

Odoo 11: Community-Enterprise Boundaries

From a product point of view, we make a difference between the Invoicing App  (invoices,
payments, taxes, reconciliation) and the Accounting App (journal entries, bank interfaces,
reporting, legal statements, budgets, assets).

We consider that the Invoicing App is part of Odoo Community (in addition to the main
apps like sales, purchase, stock, crm, project, mrp, website, eCommerce ...) and the
Accounting App is part of Odoo Enterprise (with Studio and eSign). [1]

In terms of features, the split is:
Community: Invoices, Taxes, Payments, Billing related Reports, Meta-Data needed to invoice (chart of accounts, taxes, fiscal positions, l10n_XX modules for country specific taxes and CoA).
Enterprise: Bank interfaces, Legal Statements, Accounting Reports, Meta-Data related to legal statements, budgets, assets.

[1] https://www.odoo.com/page/editions

Changes in Odoo 11

Odoo 10 Community didn't have the features expected by an accountant, and found in proprietary accounting
softwares (bank interfaces, dynamic reports, statement import, perfect localization). We never considered Odoo
Community as an accounting software, but it still had some accounting features (Chart of Accounts, Journal
Entries, Assets). This created the false expectation that Odoo Community can be used as an accounting
software (and created frustration has it does not have the quality/features expected by accountants).

To not create false customer expectations, in Odoo 11, we will promote Odoo Community as a invoicing app, but
not an accounting software. The accounting app is part of Odoo Enterprise. We do believe that a clean Invoicing app is better
than a half-accounting app.

Although the accounting engine (journal entries, fiscal positions, taxes, etc.) remains in Odoo Community
as we need them for the Invoicing app, we "hide" accounting features in Odoo Community to not create
false expectations. 

Community Modules

The above explanation is for the official Odoo distribution only. As we keep the accounting engine and most
meta-data in the Odoo Community code, it's still possible for third-party community apps to implement
accounting features based on the official Odoo code (by unhiding accounting features for example).

In short, Odoo SA does not support the accounting features of Odoo Community, but it's ok if third party
apps do it (and we don't make accounting developer's life complex).

Business Rationale

Both Odoo Community and Odoo Enterprise are important products for Odoo SA. Odoo Community
allows to attract a lot of users / contributors as it's free, and Odoo Enterprise allows to get the revenues
we need to develop the product.

Odoo SA's goal is to keep a good balance in R&D investments and spend ~80% of our resources
to improve Odoo Community and 20% to improve Odoo Enterprise. To guarantee such a ratio,
Odoo Enterprise has been designed to depend on Odoo Community so that we need to improve
Odoo Community in order to improve the overall product.

The accounting app is the main one in the ~20% because it requires much more R&D investment than others apps, especially as we have to cover a lot of countries. So it's better if these developments are financed by Odoo Enterprise in the long term (in addition, there are important direct costs to finance like bank interfaces).

Over the past years, localizations were driven by partners. But purely community driven developments didn't allow us to reach the level of quality of our competitors, proprietary accounting softwares. Even if the community did an amazing job, we were not reaching the level of Sage, Quickbooks, etc. As of v11, we decided to invest a lot in localizations, still in collaboration with partners, to develop the perfect accounting software for a lot of countries. It's a huge cost, but we expect to cover it with the extra revenues of having a clean accounting app to sell. As these modules will become part of the standard, we also guarantee evolution at each new version for customers.

**Comment by anajuaristi** [comment]
Simply a couple of point I don't agree at all
When you say:
 "We never considered Odoo Community as an accounting software... This created the false expectation that Odoo Community can be used as an accounting software (and created frustration has it does not have the quality/features expected by accountants)."
It could be true for several countries but in SPAIN we are using Odoo / OpenERP as accounting software since 5.0 and we do not create frustration nor give bad functional/technical features to customer. 
Please... there have been a lot of people involved working for free by years on this subject so I would ask to not remove the big value that people apported to community and Odoo as it is now.
If Odoo wants to run alone as you did from begining it's up to you. You never counted the thousands of programmers you have got around the word as part as your team so you wasted time and efort but again is up to you. I say nothing about that but I will not let you say that community work is not "good enough" that is "causing false expectatives and frustrations" because it's not true. It's a lie. Community has ALWAYS covered the lack of features Odoo core has for a real company, not only in accounting but in almost all the other areas. That's the real truth. If you want to kill the community work and collaboration... again is up to you but please do not kill the value, knowledge and efort of involved people because with that you are creating real frustration and disapointments. 
Thank you for your explanation:
Ana

**Comment by yostashiro** [comment]
It's sad that we can no longer promote Odoo as open-source ERP...  And @lap-odoo probably should have clearly explained the intention/background from the beginning when these localization threads are created.

**Comment by Yenthe666** [comment]
I agree with @anajuaristi and @yostashiro here.
This somewhat feels (atleast to me) like tricking the community into sharing knowledge, helping and testing the development. I was under the assumption that this would still be for the community too and this made me tweet with false assumptions too. I honestly believe that Odoo will be stepping on quite some toes with this one, this could have been handled better with clear communication.
P.S Please still add this on the initial message of every issue for localizations. This should be clearly communicated on every bug report.

Regards,
Yenthe

**Comment by fhe-odoo** [comment]
@anajuaristi, I'm pretty certain that @lap-odoo was exclusively talking about the official modules contained in the Odoo Community edition and not about the community modules.
She even wrote "The above explanation is for the official Odoo distribution only."

Also, she wrote "it's still possible for third-party community apps to implement
accounting features based on the official Odoo code".
- Who talked about removing the value brought by the community modules?
- Who said that the work of the community is "not enough"?
- Who talked about killing the value, knowledge, and effort of the community?

Did you read the whole message?

**Comment by fpodoo** [comment]
I confirm what Fabrice said: "Laura's message is purely about the official
distribution". It's not a critic of the community, or community modules.

I actually wrote this message myself in another context; to explain why we
removed the account_accountant module from master to OCA, not related to a
specific country.


We consider that the official distribution "Odoo Community", in standard,
doesn't have the accounting quality expected by accountants. So we prefer
to not advertise it that way.

Of course, we know that some countries, including Spain, have community
modules that compensate missing features. And we do respect and value these
community modules, that's why we kept the whole accounting engine, in Odoo
Community standard. (So you can re-show all accounting features in your
community module, by setting a group, and have the standard CoA, taxes,
accounting engine...)


In particular, for l10n_es contributions, if I am not wrong, most feedback
(if not all?) were about CoA, taxes, meta-data and have been merged in Odoo
Community.

One is border line: the tags. If I am not wrong, the tags are in Odoo
Community meta-data but are used by Enterprise reports. Sorry, if we
haven't clarify this earlier. Our intentions are certainly not to trick
you. I guess we did not reexplained it as we probably thought you knew it
as it's like that since v9 and v10.

Our approach tries to satisfy everyone needs and take into account every
actors interest; community modules need an accounting engine they can rely
on, community members want to contribute to Odoo Community mostly, be clear
for end-users about whats in the product "Odoo Community" that they
download, the need to protect the value of Odoo Enterprise to sustain the
project financially.

Hope it clarifies the border between the two products and why we think that
way.

**Comment by pedrobaeza** [comment]
Thanks all for clarifying, although answers being bitter, so coming back to the issue itself, as stated I will continue giving advise/review for free on the Odoo community part.

@oco-odoo About the account types, they were kept for legacy reasons (although I clean/merge some of them in one of the refactoring), but there's no real reason for keeping them as they are. The use of that types given by Odoo in version <= 8 was for the balance sheet and profit & loss and you know that we have our own BS/PL reporting engine, so it was not used on that versions nor the newer ones, so you can remake them as your convenience.

**Comment by cmsalmeida** [comment]
@pedrobaeza I hope this doesn't seems to be only the beginning...
Of course all code customised and developed by the community should remain CE. But more than that receive the proper merit and respect they deserve by at least avoid paths that would create incompatibilities as much as possible. Share is free, but development and maintenance costs a lot of time, money and effort from all of us. A simple math would reveal millions across the globe for sure. 
I share my concerns with @jbeficent @lesley @liebana @yostashiro and @pedrobaesa. Not only on localizations but on other modules as well. After all Odoo is supposed to always be open source. 
It will be very interesting to see this for l10n_br one, even if one start from one of the two workable versions. It would be a nightmare. 
But as @lap-odoo said as a community, partners or developers we can always maintain and use our own implementations. Even if we need to fix the created incompatibilities or re-develop removed features. Of course all contributions possible to existent localizations would very appreciate and, will  have a better cost per added value relation.
Good to have an OCA.

**Comment by rosenvladimirov** [comment]
Removing account_accounting is a very good idea; in my opinion, it is necessary to re-structure the installation of the package. By re-grouping business logic modules and base modules. Moving business logic modules into the app folder will "unlock" the replacement options with more advanced modules. In the 10-version, this separation is observed with the enterprice modules, but in my opinion this should be done as a whole. It is a necessary engine to manage the replacement of core business logic modules. Currently upgrading of features and modules is not effective enough. With this system expansion model, the processing time of a given logic increases. The system goes through all the installation modules. I think it's very difficult already to track the basic flow of logic movement after installing multiple add-ons (modules) over a primary one. In this direction, steps need to be taken. In terms of localization: it is rather a tuning of a sufficiently flexible module than the rewriting example of account_aconting. Globalization currently removes the local peculiarities of accounting. In the future, we will see the definitive abolition of national accounting standards and their replacement with international accounting standards. with regard to taxes, several major areas with different tax varieties are formed in the world. Local taxes will be just rates, not models.

**Comment by Ander-tc** [comment]
Hi Laura,

Regarding the accounting models that we are currently working on:
- Models 200 and 202 are mandatory for every company (confirmed by the "Agencia Tributaria") and should therefore have a high priority.
- Models 111, 115, 303 are being currently targeted. Since models 190, 180, 390, are annual summaries of the previously mentioned ones, I believe that we should have then ready shortly after so that companies can generate them through Odoo by January 2019.

**Comment by jalzaga** [comment]
That´s right 
Model 200 (Corporation tax) is mandatory in Spain, it should be presented from June 1st to June 20th 
Model 202 (Split payment for corporation taxes) is also mandatory an it should be presented in April, October and December

**Comment by js-landoo** [comment]
@ander-tc and @jalzaga , tax forms 200 and 202 are mandatory but to be filled they need some data not present in Odoo, so I think it's not expected to be done in Odoo.
Our customers do it once a year and I'm sure they will prefer to continue with excel.
The proof is that actual spanish localization, doesn't implement 200 nor 202.

I think we have to concentrate our/Odoo's efforts in really expected features.

**Comment by Ander-tc** [comment]
Hi @js-landoo , thanks for the information. Could you please clarify what information is necessary but not present in Odoo?

**Comment by js-landoo** [comment]
@Ander-tc , have a look to this pdf (738 pages) lo learn how to calculate it :-)
http://www.agenciatributaria.es/static_files/AEAT/DIT/Contenidos_Publicos/CAT/AYUWEB/Biblioteca_Virtual/Manuales_practicos/Sociedades/Manual_Sociedades.pdf

Basically you need Balance and Profit&Loss, but many information for about 20 different tax reductions (employees and contract types, investments, ...), company type (many attributes) for applicable tax percentage, an so on.

**Comment by lap-odoo** [comment]
@js-landoo : Thank you for this explanation. 

I agree, I think we should first focus on reports that really have an added value for the users if they are generated from Odoo. 

@pedrobaeza : Do you agree with this too ? 

The improved Spanish Localization has been merged in SaaS 18. So feel free to test (community and/or enterprise version) and to report issues. Thanks

**Comment by pedrobaeza** [comment]
Indeed that models, although required are too complex to be introduce right now in Odoo, because it requires a lot of data not present directly on the transactional information (for example, you will need a mechanism for identifying the journal entry that distributes the gains), or to enter again all that information. That's why we haven't even done it on OCA/l10n-spain.

Most people goes to an external adviser for this or uses an spreadsheet for this task.

**Comment by lap-odoo** [comment]
@pedrobaeza Thanks for your quick answer

**Comment by jalzaga** [comment]
The point is not the complexity, everything can be achieved regardless the data is not 100% in odoo, there are many ways to enter the additional data, other companies such as Sage has done it.  We can focus on the reports that are more urgent in the timeline but we should not disregard the other ones, otherwise it will be always the same topic: Odoo does not cover the Spanish legislation and other softwares do.  @lap-odoo

**Comment by lap-odoo** [comment]
@jalzaga I agree, I expect to have these reports in Odoo at some point but we must focus on the priorities first : reports that have the biggest added value if done in Odoo (since all the required data are already in the database).

**Comment by pedrobaeza** [comment]
I agree with @lap-odoo. Spanish localization is there since TinyERP 4.2, and these reports have never been a blocker thing for any implementation.

**Comment by anajuaristi** [comment]
+1 @lap-odoo @pedrobaeza

**Comment by gelo-landoo** [comment]
Hi @lap-odoo ,
I have been watching saas18 and I have detected one thing that could be a problem.
![01](https://user-images.githubusercontent.com/25029267/30634142-5eb4eb9a-9dee-11e7-970e-226d49a53e13.png)
On accounting configuration you can only add one creditor identifier (1) and one name identification (2). Spain banks has each one its own SEPA data. If you are working with three or four different banks,  I think it´s not the best way to work having to be changing this data every time.
I think that, for example, each bank journal could have their own data.

**Comment by oco-odoo** [comment]
Hi @gelo-landoo ,

I am the one who implemented SDD, so I can answer for point 1. In SDD, the creditor identifier uniquely identifies the customer (that is, your company) in the scheme. It is thus *in no way* related to your banks. It stays the same whatever the bank accounts you use and the banks you work with. So, we just need one field in the options.

SCT in the other side of the scheme, and as I did not code the module related to it, I'm not sure my remark also applies.

**Comment by gelo-landoo** [comment]
Hi @lap-odoo  @oco-odoo 
I'm testing Saas_18 and I have detected other subjects that I think would be better to be discussed.
To explain them, screenshots and different point of view are needed, so this issue maybe is not the best place, can became quite confusing.
Just let me know which do you think is the best way to contribute.
Best regards,

**Comment by lap-odoo** [comment]
@gelo-landoo : If it's not directly related to Spanish Accounting you can use this issue to report your remarks : https://github.com/odoo/odoo/issues/18637. If the discussion becomes too complex, we'll switch to another channel but at least the other partners can give their opinion about your suggestions. 

With the Odoo Days coming, I didn't have the time yet to have a look at all the remarks on this issue but I'll take care of this once the Odoo Days will be behind us ;)

**Comment by gelo-landoo** [comment]
Hi @oco-odoo @lap-odoo 
I´m not an expert on this subject, @pedrobaeza  knows way more than me and he can correct me. In OCA was made a PR on the module that is used to generate SEPA operations. And in the file "Readme" it  was said:
"Account Banking PAIN Base Module
This module contains fields and functions that are used by the module for SEPA Credit Transfer (account_banking_sepa_credit_transfer) and SEPA Direct Debit (account_banking_sepa_direct_debit). This module doesn't provide any functionality by itself.
This module was started  [...]
Configuration
Go to Accounting > Configuration > Settings.
On the fields "Initiating Party Issuer" and "Initiating Party Identifier", in the section SEPA/PAIN, you can fill the corresponding identifiers.
**If your country requires several identifiers (like Spain), you must:**
Go to Accounting > Configuration > Settings.
On the section SEPA/PAIN, check the mark "Multiple identifiers".
Now go to Accounting > Configuration > Management > Payment Modes.
Create a payment mode for your specific bank.
Fill the specific identifiers on the fields "Initiating Party Identifier" and "Initiating Party Issuer"."
I´ll give you the link: [https://github.com/OCA/bank-payment/blob/10.0/account_banking_pain_base/README.rst](url)
May be I am wrong and only one creditor identifier is valid, but think it isn´t. If this PR was created it would be because it was necessary.
Best regards,

**Comment by pedrobaeza** [comment]
All of this discussion should be placed on enterprise branch, as Odoo SEPA support is enterprise.

**Comment by Elbagoury** [comment]
"We never considered Odoo Community as an accounting software... " 
I don't agree with that in the Arab countries especially in Egypt we are using Odoo / OpenERP as accounting software and  this way of thinking will make Odoo SA lose thousands of customers

**Comment by felixvillafranca** [comment]
Hello Laura;

Thanks for your work¡¡¡

I would like to know what's the situation for the next modules V.11:
- SII, mandatory for some companies in Spain.
- Model 180, 190, 390, 347 y 349

And, what about "l10n_es_account_invoice_sequence" (V11 thanks to OCA) and "l10n_es_partner" (V10)? These modules are too mandatory in spain, will they be included in the core of Odoo? Or conversely, we will have to install as submodules. 
In my opinion this is a bit confusing; and could be interesting to have a roadmap about what modules are going to be supported, for whom and when they will be availabe.

Best regards,

**Comment by oco-odoo** [comment]
Hello @felixvillafranca ,

I am currently working on phase II of Spanish localization, so I can tell you what it consists of. Here is a list:

- Mod 347 report
- Mod 349 report
- export to BOE format for every implemented mod report (111, 115, 303, 347, 349)
- N43 importation
- SII

I have already implemented mod 347 and 349, as well as BOE for mod 111 and mod 115. I still have a few things to fix on them, and am now working on BOE for mod 303. You will soon be able to test those.

**Comment by felixvillafranca** [comment]
Hello @oco-odoo ,

Perfect, great job ¡¡¡

What about the other question; "l10n_es_partner" and "l10n_es_account_incoice_sequence" have to be installed as submodules? or will they be included?

Regards,

**Comment by lap-odoo** [comment]
Hi @felixvillafranca, 

What do these modules contain? Could you explain us what is missing in Odoo standard? Thanks. 

@js-landoo @pedrobaeza  : If I'm correct, we have not talked about these modules yet, is there a reason for that ? Thanks :)

**Comment by jalzaga** [comment]
Hi @lap-odoo , l10n_es_partner provides information about Spanish banks and drives and validates bank account codes (IBAN). I think it is very good to have.
Regarding the l10n_es_account_invoice_sequence, what it does is make invoice number sequence different from journal number. In my view it is not necessary since you can keep invoice sequences without gaps. The only need in Spanish accounting is to have a sequential number for all the account moves regardless the journal they belong to, this could be done whit a simple module. @felixvillafranca @oco-odoo @liebana

**Comment by felixvillafranca** [comment]
Hi;

In addition to what @jalzaga says, "l10n_es_partner" add a  the field "Trade name" to res.partner when are companies, of widespread use in Spain.

In relation to "l10n_es_account_invoice_sequence", I'm not accountant, maybe @jalzaga is right. But why you can read in the [README](https://github.com/OCA/l10n-spain/tree/11.0/l10n_es_account_invoice_sequence) that its use is mandatory for Spain?

Regards,

**Comment by fpodoo** [comment]
@lap-odoo From what I remember, "l10n_es_account_invoice_sequence" exists for historical reasons, but Odoo his better without it than with it. A few years ago (v7, I think), we had two different sequences for invoices and journal entries. When we decided to unify both sequences, the community developed this module to keep the old behavior.

In reality, having two different sequences makes everything more complex: reports should display both sequences to allow easily matching of data, it's more complex for the user (as the invoice does not have the same reference than its journal entry), etc. And it's certainly not required by the law. So, it's better to not include this module in standard.

**Comment by lap-odoo** [comment]
@jalzaga @felixvillafranca Thanks for the info.

About the validation of the IBAN bank accounts, this already exists in Odoo standard (module: base_iban). 

What's the use of the "Trade Name"? Is it used when the company name(=trade name) is different than the account holder name (=official company name)?

**Comment by jalzaga** [comment]
@felixvillafranca @lap-odoo In Spain the only think that is mandatory is to have a sequential number for invoices without gaps. This is something that odoo already provides. 
In addition, the Spanish law obliges to report at the end of the year all the account_moves with a unique sequential number without gaps, regardless the journal they belong to. This has been managed by the community by renumbering the account moves at the end of  the year, but in my view this is confusing, so it would be better to add as sequential number a new field different from the account_move name. @fpodoo

**Comment by jalzaga** [comment]
@lap-odoo @felixvillafranca Yes, the trade name is very common, especially for shops, there is a trade name (commercial) such as, "Rick's", and the official name to report legally is the name of the owner "Ricardo García Fernández'

**Comment by felixvillafranca** [comment]
@lap-odoo @jalzaga, that's right.

**Comment by felixvillafranca** [comment]
@fpodoo thanks for the explanation, I agree with you, If it is not necessary it is better to leave it as simple as possible.

**Comment by anajuaristi** [comment]
@fpodoo Just to say that l10n_es_account_invoice_sequence is not manteined for "historical reasons" but "legal requirement". 
In spain is a legal requirement:
.- For invoices: It's right they are numbered by journal
.- For account move: Single number sequence ordered by data and without  "holes" between numbers
The module is just adding a field sequence on journal, but it's not necesary a new field sequence on account move as @jalzaga is proposing. That would be confusing I think.

**Comment by fpodoo** [comment]
> @fpodoo <https://github.com/fpodoo> Just to say that
> l10n_es_account_invoice_sequence is not manteined for "historical
> reasons" but "legal requirement".
> In spain is a legal requirement:
> .- For invoices: It's right they are numbered by journal
> .- For account move: Single number sequence ordered by data and without
> "holes" between numbers
>
Do you mean that, in Spain, you can not have a sequence of the journal
entries per journal? Do you have a reference to this law?


-- 
Fabien

**Comment by jalzaga** [comment]
@fpodoo In Spain, legally there is only one joutnal, it is called libro diario, it takes all the account moves for the whole fiscal year in chronological order. Odoo provides different sequences for each journal. But  there is not a problem to use Odoo standard, the only thing we have to do is provide a report for the legal authorities with a chronological number.   
Regarding the invoices, Odoo is completely complaint wiht Spanish Legal requirements since it provides a unique, sequential number without gaps. @lap-odoo

**Comment by mreficent** [comment]
In OCA, we have this _libro diario_ called Journal Ledger. See https://github.com/OCA/account-financial-reporting/tree/9.0/account_journal_report. @fpodoo It could be *super*-nice if Odoo adds this report in the [account reports](https://github.com/odoo/odoo/tree/master/addons/account/report).

**Comment by etobella** [comment]
Well, _l10n_es_account_invoice_sequence_ is necessary for two reasons:
- In Spain, invoices must be correlated in a series and exercise. If you use the standard, gaps could be created if you create a manual move on the journal. It shouldn't happen but mistakes exists.
- Refunds must have a different series than the original invoice (and therefore a different numeration)

So it is necessary to have 3 sequences on sale invoice journals: one for account moves, one for invoices and one for refunds. 
I also think, that it could be interesting to split the invoice number between number and series.

**Comment by felixvillafranca** [comment]
@etobella , About refunds, as far as I know you can define in the sales journal if you want a diferent sequence for credit notes.

**Comment by etobella** [comment]
@felixvillafranca you may change the journal, but I think that the system should not allow you to make mistakes. As far as I am concerned, I must find the way to minimize errors.
Maybe it is not necessary the third sequence, but then, the system should not allow you to reuse the same journal that is used in the original invoice.

**Comment by felixvillafranca** [comment]
@etobella IMO If you set up the journal properly, it is not necessary to change the journal to register a credit note.
![screenshot 2017-11-06 at 22 36 25](https://user-images.githubusercontent.com/4075475/32465204-fc2dce18-c342-11e7-85f2-0ba5666888dd.png)

**Comment by etobella** [comment]
@felixvillafranca my mistake, you are totally right. Refunds may have its own sequence.

**Comment by jalzaga** [comment]
@etobella the  possibility to enter an account move using the invoice journal may happen rarely but, in my view, this is not a reason to change the Odoo standard and make it more complex, as @fpodoo said.

**Comment by lap-odoo** [comment]
Could someone provide me the law that explain the requirements for this sequence number for the accounting moves ? Thanks

@etobella : Indeed, in standard you can have a different sequence for the refund 

@jalzaga @felixvillafranca : regarding the trade name, what we will do is to add a field on the bank accounts (res.partner.bank) where to enter the exact name of the bank account holder. If the field is empty, the field partner_id will be used. If the field is filled in, the information will be used to generate the SEPA files for example. Example : Trade name = Rick's but the name of the bank account holder is "Ricardo Fernandez", you will be able to enter that info on the bank account directly. 

So, if I'm correct, the only thing that is left to discuss is this numbering on the account moves (no matter what the accounting journal is).

**Comment by felixvillafranca** [comment]
@lap-odoo , IMO what you propose is not correct, this field is linked to res.partner not to res.partner.bank.
You can take a look to [OCA module](https://github.com/OCA/l10n-spain/blob/11.0/l10n_es_partner/models/res_partner.py). Notice that is important to provide a way to search using this "Trade Name".
Regards,

**Comment by lap-odoo** [comment]
@felixvillafranca : What we suggest is to enter the trade name as the normal name of the partner. That's the name that will be used the most often (the "common" name). From what we know, the only time where the "official" name is needed is when making payments, since the bank can require that official name linked to the bank account. Are there other use cases where you think this has an implication ?

**Comment by felixvillafranca** [comment]
@lap-odoo , Don't forget that in adition to the paymentes you have to present all reports (legal reports) with the legal name.

**Comment by lap-odoo** [comment]
@felixvillafranca Do you mean your legal reports (your P&L for example, you want to see your official name) or do you mean that when you send an invoice to a customer you want to address it to its official name ?

**Comment by AngelCamacho** [comment]
@lap-odoo  In spain Invoices , P&L have to see the official name (legal name), not only the payments. The trader name it's only for comercial suposes and it's a easy way to find a customer with his trade name.

**Comment by lap-odoo** [comment]
@AngelCamacho "It's an easy way to find a customer" > I agree, that's why the trade name should be entered as name of the partner_id. 

P&L : You only use it internally, right ? When sending this information to the Government you have to re-enter the data (and the official name of the company), isn't it ? 

On the invoice : do you mean you need to send it to the official name of your customer or using your official name ?

**Comment by felixvillafranca** [comment]
@lap-odoo , It's mandatory to show your official name and your VAT number in the invoice, although you can also show your Logo & Trade Name.

**Comment by felixvillafranca** [comment]
@lap-odoo , check [the law](http://www.boe.es/buscar/act.php?id=BOE-A-2012-14696&p=20141220&tn=1#a6) (in spanish sorry).

**Comment by lap-odoo** [comment]
@felixvillafranca : Point C I guess ? I'm sorry I don't really see where the make the distinction between trade name and official name, Google Translate is probably not precise enough. Could you please precise which point of the law I should be looking at ? Thanks

**Comment by gelo-landoo** [comment]
@lap-odoo In point C where it´s said " Nombre y apellidos, razón o denominación social completa" means the official name. "Nombre y apellidos" is for example a freelance (Miguel Ángel García Rodríguez). "Razón o denominación social completa" is the official name of the enterprise, as it has been registered, for example "Vodafone España, S.A.

**Comment by felixvillafranca** [comment]
@lap-odoo , Try [this](https://www.oepm.es/es/propiedad_industrial/preguntas_frecuentes/FaqSignos01.html)

**Comment by oco-odoo** [comment]
Hello everyone, 

As promised to @felixvillafranca a few days ago, I just finished a subset of Spanish localization's phase II's features, and you can now test them :) 

You can find the dev branch on runbot, in odoo-dev/enterprise, under the name master-l10n-es-2-oco. Please keep in mind that this is all a work in progress, so it is possible that, sometimes, some commits are red on the branch, and some usability improvements are likely to happen in the near future. We prefer to give you an early look at the branch in case we made a mistake somewhere, so we are sure to fix it in time ;)

To summarize, here are the features you can test for now:
- Mod 347 report (but not its BOE)
- Mod 349 report (but not its BOE either)
- BOE for mod 111, 115 and 303

I'll now be working (in this order) on mod 347 and 349 BOE, N43 and SII. I'll let you know as soon as it's done ;)

**Comment by felixvillafranca** [comment]
Hi @oco-odoo ,
I'll test it ASAP.
Thanks for your great work ;·)

**Comment by felixvillafranca** [comment]
Hi @oco-odoo ,

IMO 347 report should follow the AEAT's structure, that you can find [here](http://www.agenciatributaria.es/static_files/AEAT/Contenidos_Comunes/La_Agencia_Tributaria/Modelos_y_formularios/Declaraciones/Modelos_300_al_399/347/Instrucciones/Instr347_2010.pdf).

Basically, there should be a summary:

- Total number of people and related entities in the inner sheets.
- Total amount of related operations in the inner sheets.
- Total number of related properties in the attached lease sheet business premises
- Total amount of related operations in the attached sheet of lease of business premises

And then the inner sheets;

- List of declared, notice that amount would indicated by Quarter. 
- List of properties, if needed

I know this is not BOE, but the structure should be the same, perhaps with less detail.

Regards,

**Comment by lap-odoo** [comment]
Hi,

Regarding the trade name vs official name, this is in fact also valid in Belgium. This is not specific to Spain. For your own company, you can simply use the custom report footer to display also your official name. For your customers, you could for example use an invoicing address that is the official name of the company. The field street2 could also be used. 

We want to avoid creating new fields to manage all the peculiar use cases that can happen, especially if workaround exist. 

Do you have a link to the law about the unique reference shared between all accounting entries ? 

Thanks

**Comment by felixvillafranca** [comment]
Hi Laura;

The ways you exposed for handle this could be enough, but maybe for some people could be a little twisted and not efficient. 

Because the trade name is something really widespread in Spain I think there should be a unique and clear way of use it and not a lot of workarounds.

Thanks,

**Comment by mart-e** [comment]
An improvement of the l10n_es module has been submitted at #20927 by @rlizana.
Reviews are appreciated.

**Comment by pedrobaeza** [comment]
Yeah, it's one of the results of the Madrid code sprint that restores the PGCE account hierarchy through the new model account.group.

About the trade name, the approach given by @lap-odoo is about company, but what `l10n_es_partner` module does is for every res.partner: each customer/supplier can have a trade name, and you want to search them by that trade name. Example: the fiscal name (which should appear in invoices) for one partner is "Alcoholera Vínicola de España S. A.", but the trade name is "Alvinesa", and you want to type "alvi.." and find it.

**Comment by jbeficent** [comment]
@lap-odoo @pedrobaeza I have created https://github.com/odoo/odoo/pull/21170 to change the account_type of 129 to unaffected earnings.

**Comment by oco-odoo** [comment]
Hey everyone, 

I need some help to understand the BOE export file structure for mod 347.  

Page 24 of the documentation (http://www.agenciatributaria.es/static_files/AEAT/Contenidos_Comunes/La_Agencia_Tributaria/Ayuda/Disenyos_de_registro/Ayudas/DR_300_a_399/ficheros/347.pdf), three fields are defined:

1) Operaciones régimen especial criterio de caja IVA
2) Operación con inversión del sujeto pasivo
3) Operación con bienes vinculados o destinados a vincularse al régimen de depósito distinto del aduanero

(I'll call them field 1) to 3) from now on)

I have a few questions about these

- Can these fields be 'True' (have value 'X') together ? To me, it would make sense that they are mutually exclusive; am I wrong ? 

- Field 1) is used to mark operations for which we pay IVA when we have collected the money, right? So, am I correct if I say the field defined page 25 (importe anual de las operaciones devengadas conforme al criterio de caja del IVA) only contains the amount that has already been collected? (i.e., the amount for which IVA must be paid)

- I don't understand the purpose of field 2) :s Can anybody provide an example ?

- Field 3 indicates the operation follows a specific fiscal regime. In what conditions can this happen, in practice ?


These fields were supported by the community module, but they prompt the user to enter everything manually, with a very specific modelling that I cannot replicate on enterprise (for consistency in accounting reports), so I need to fully understand everything so that I can re-implement it 'the enterprise way'.


So... any help would be appreciated! Thanks in advance :)

**Comment by lap-odoo** [comment]
Hello, 

One PR has already been merged (#21170) and the other one will be reviewed (#20927). 

@pedrobaeza : You can use the same for the res.partners. Create a res.partner with the trade name and link it to an invoicing address with the official name. You could search on the two information easily.

**Comment by pedrobaeza** [comment]
@lap-odoo thanks for the trick, but this is not what we try to achieve. In my example, you want to type "alvi..." and find all the contacts related the main company that has the trade name "Alvinesa", not only a contact that has this alternate name.

**Comment by lap-odoo** [comment]
@pedrobaeza I understand this is not ideal, however, I don't think managing such use cases should be a priority for us right now as (1) these are exceptions (2) workarounds exist and (3) it's easy to develop a quick solution. I prefer to use our resources to work on the different tax reports for Spain for example. 

Could someone help @oco-odoo with his question about the BOE file ? 

Thanks

**Comment by pedrobaeza** [comment]
@lap-odoo OK, as you prefer.

About @oco-odoo's concerns:

* `1. Operaciones régimen especial criterio de caja IVA`: This is cash basis criteria, so I think you can obtain data through standard cash basis.
* `2. Operación con inversión del sujeto pasivo` This is a special tax criteria for certain operations, where the partner of the invoice is the company itself. We are checking this right now manually on the community module, but it can be automated.
* `3. Operación con bienes vinculados o destinados a vincularse al régimen de depósito distinto del aduanero` This is another distinction made by AEAT for the nature of certain operations. I don't even use it any time and don't know it very well. It's for the goods you leave on the customs house You can check http://www.sunat.gob.pe/orientacionaduanera/cartillasorientacion/cartillasProcedim/tr01Depo.pdf. We don't have anything (taxes nor check) for distinguish it. You can leave it non covered and it will be valid for the 99% percent of companies using Odoo.

**Comment by oco-odoo** [comment]
Hi  @pedrobaeza, and thank you for your answers.

So, to get on with all this ...

1) Alright. But how do I report it in the file ? Should I write all the operations regardless of the use of cash basis, and then generate other records in the file with cash basis amounts ? If so, should I only put the collected tax amount in these records, or the total amount of the operation ? 

2) Okay, so I'll just test that the partner related to the record is the same as the company's.

3) For this, maybe we could just add a boolean field on invoices ? I agree it is not a must-have if nearly nobody uses it, but this little workaround would allow the few ones who do to still use Odoo properly.


I have other questions, I'll make a list and ask them as soon as we're totally clear with this one ;)

**Comment by oco-odoo** [comment]
Hello,

Just to let you know that I'm clear with the development of mod 347 and 349 BOE files ! 

As I told Pedro the other day, I just have a few questions left. So, some fields of the mod 347 BOE file are blank for now, as I need to be sure of their content. Apart from this, you may test everything! :) Latest build on runbot is here http://287734-master-l10n-es-2-oco-b6245e.runbot13.odoo.com/web#home .

I already asked some questions in my last message. Here are some more: 

4. In Mod 347 BOE file, I don't understand what must go in the "Importe Percibido en Metálico" field of the partner records. Any clue ?

5. In Mod 347, again on partner records, what are the fields about "inmuebles sujetas a IVA" for ? In what way do they add something to the "registros de inmuebles" that are located further on in the file ? What should I put there ?

6. In Mod 347, on partner records, there is a field called "NIF Operador Comunitario". Is this field mandatory for all partners inside the EU, or is it just some additional info that can be specified if we have the TIN (NIF) of the partner ? As it is now, I'm considering this field as mandatory, so it is impossible to generate a BOE file if any of the customers is located within the EU and does not have any TIN set. I think it's super-restrictive, so it is important to know if this field is mandatory or not, so that I can remove the check if it is not. 


I hereby summon the power of the community to solve these questions !! ... No, seriously, a little help would be very welcome :) Once we have solved those questions, our Spanish localization base package will be complete, so it's really the final stage of development. :)

Thanks in advance !

**Comment by gelo-landoo** [comment]
@lap-odoo Where can we test the latest versions of the l10n_es?

**Comment by lap-odoo** [comment]
@gelo-landoo It's the branch called master-l10n-es-2-oco on our [runbot](http://runbot.odoo.com/runbot/repo/git-github-com-odoo-dev-enterprise-8?search=). I've just launched the rebuild of the branch so you'll have to wait a bit (+/- 45 min) before being able to connect (admin, admin).

**Comment by FDiaz-EnovaTICs** [comment]
Hello to every one,
It would also be convenient to add Canary Island's taxation in considerations.
Although most companies do not use V.A.T., some do, we share some of the modules with Hacienda Nacional.
IGIC is the canary island taxation behalf of the VAT and we use MOD420 for every quarter and 425 for every year declaration among other regional forms. http://www.gobiernodecanarias.org/tributos/portal/jsf/publico/asistenciaContribuyente/modelos/inicio.jsp
http://www.gobiernodecanarias.org/tributos/portal/jsf/publico/asistenciaContribuyente/modelos/listado.jsp?tributo=IGIC&jftfdi=&jffi=listado.jsp
Regards

**Comment by FDiaz-EnovaTICs** [comment]
Hello,
Also would convenient to see the SSI  (Suministro Inmediato de Información)
Technical details http://www.agenciatributaria.es/AEAT.internet/SII.html
BOE Links http://www.agenciatributaria.es/AEAT.internet/Inicio/La_Agencia_Tributaria/Campanas/Suministro_Inmediato_de_Informacion_en_el_IVA__SII_/INFORMACION/Normativa/Normativa.shtml

It seems that Canary Island, as far as I know, will not be implemented yet as we are on IGIC scheme.
Regards

**Comment by lap-odoo** [comment]
Hello, 

I have a question regarding "inversion del sujeto passivo". Should it be included in the base imponible or not? In the tax report of version 10, inversion del sujeto passivo is subtracted from the base imponible but we are wondering if it shouldn't be added to the base imponible. 

That's what I understood from [here](https://www.bbva.es/general/finanzas-vistazo/empresas/inversion-sujeto-pasivo-iva/index.jsp) but I'm not 100% sure the Google translation can be approximative. 

Thanks for your help

**Comment by lap-odoo** [comment]
@FDiaz-EnovaTICs Implementing SII is well part of our roadmap. We'll first finish the implementation of the modelo 347 and 349 before starting to work on SII.

**Comment by dipi44** [comment]
@oco-odoo 

I will try to help with some questions here, though I am not a tax advisor.

Q4. In Mod 347 BOE file, I don't understand what must go in the "Importe Percibido en Metálico" field of the partner records. Any clue ?   

A4.   

Background

In Spain there is a limit on B2B and B2C cash transactions. All invoices that are above 2500 euros cannot be paid in cash.   This limit is increased to 15000 euros for C2C (non business entitities) and B2C(Non Tax-resident customers). 

http://www.elderecho.com/tribuna/fiscal/limitacion-pagos-efectivo-normativa-antifraude_11_486430001.html

Reference to the law:  Please refer to Article 7 on page 76285

https://www.boe.es/boe/dias/2012/10/30/pdfs/BOE-A-2012-13416.pdf




Therefore in Modelo 347 all cash transactions above 6000 euros must be reported for all business partners (Customer and Supplier)  If there are no cash transactions greater than 6000 euros but the sum of individual transactions  is greater than 6000, it must also be reported.  

This provides valuable information to the spanish Tax authorities to detect tax fraud, and it can keep an eye on companies to ensure they are complying with the 2500 euros limit rule on cash transactions. (Smurfing / Artificial fractioning of invoices  is also prohibited in Spain, as fractioning a invoice into several invoices if it is detected by the tax authorities will incur in heavy penalties and legal consequences)





Example 1 : Customer1,  Transaction 23:  Paid in Cash 4000 euros, Transaction 24: Paid in cash 1500 euros Transaction 29: Paid in Cash 1000 euros.  

Result: Although Customer1's individual transactions do not exceed 6000 euros the sum of total cash received from customer1 exceeds 6000 euros and must be reported.   "Importe Percibido en Metalico"= 6500,00 euros.  

Please refer to this document, it might result in a better automatic translation.  http://www.fiscal-impuestos.com/files-fiscal/Disenos_registro.347.pdf

Reference to the law:  

https://www.boe.es/buscar/act.php?id=BOE-A-2007-15984&p=20171230&tn=1  (6000 euros 347 section 1F)

In my personal opinion, as number one priority and even greater priority than tax reporting,  I believe that Odoo Standard must provide an easy way in which this law (limiting cash transactions to 2500 euros per operation as It is illegal in Spain to pay in cash more than 2500 euros to a business partner since 2012) can be complied in Spain as it can have serious legal consequences for both the company and its employees.   

Accroding to the law this limit is per operation and not per invoice.   Example:  A customer orders 20 Iphone X with a total order value of 20000 euros. This order is divided into 10 invoices i.e. 1 every day.  The maximum amount that the customer can pay in cash for the the whole operation is 2500 euros, ie, the sum of cash payments for the 10 invoices of 2000 euro each cannot exceed 2500 euros in total.   A easy work around for this must be found, because if an employee  makes a mistake (which is very common), the company will incur in hefty fines. I know..... spanish tax authorities are not making the rules that simplify business processes, but totally the opposite.  Tax reporting can be done with third party accounting software, Sage A3, however this restriction needs to be on a day to day operational level embedded in Odoo standard.  Maybe there is a easy work around for this. If there is please let me know.

Thanks.

**Comment by pedrobaeza** [comment]
Should this be closed now that the features have been developed?

**Comment by Yenthe666** [comment]
@pedrobaeza I believe they should all remain open, in most languages things are still moving/improving.

**Comment by gelo-landoo** [comment]
Hello @lap-odoo. On What about the Spanish Profit and Loss? There wasn't on V.11 and it is not included in V.12.
This report is such important as Balance Sheet.

**Comment by lap-odoo** [comment]
@avw-odoo When you have some time, could you please have a look @gelo-landoo 's request ? thanks

**Comment by avw-odoo** [comment]
@JCRodoo Can you have look here ?

**Comment by gelo-landoo** [comment]
@JCRodoo hello,
I have in CSV format the Balance sheet and Profit and Loss for PGCE PYMEs 2008 account chart and for the PGCE completo 2008 account chart. I don't have the other two account chart models. I did it for V.10 and V.11.
If you give an email address I can send it to you.
Regards

> @gelo-landoo Hello,
> for the moment there is no development planned, but if you have some information input for me to get this dev moving, I would be glad to help and to see what I can do to get this done.
> The best case would be a pull request that we can review !
